coordinator-common/src/main/java/org/apache/kafka/coordinator/common/runtime/CoordinatorRuntime.java

```diff
@@ -197,6 +199,11 @@ public Builder<S, U> withExecutorService(ExecutorService executorService) {
             return this;
         }
 
+        public Builder<S, U> withAuthorizer(Optional<Authorizer> authorizer) {
```

Comment:
It should not go via the runtime. You can directly pass it to the shard.

Reply:
Check here: https://github.com/apache/kafka/blob/trunk/group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupCoordinatorService.java#L196